__tests__/restful-model-collection-spec.js

```diff
@@ -515,6 +538,65 @@ describe('RestfulModelCollection', () => {
       });
     });
 
+    describe('test folder and label deletion', () => {
+      test('folder should not have a delete request body', done => {
+        testContext.collection = new RestfulModelCollection(
+          Folder,
+          testContext.connection
+        );
+        testContext.item = new Folder(testContext.connection, {
+          id: '123',
+          displayName: 'foo'
+        });
+        testContext.collection.delete(testContext.item, {'display_name': 'not required'});
+        expect(testContext.connection.request).toHaveBeenCalledWith({
+          method: 'DELETE',
+          qs: {},
+          body: {},
+          path: '/folders/123',
+        });
+        done();
+      });
+
+      test('label should have a delete request body', done => {
+        testContext.collection = new RestfulModelCollection(
+          Label,
+          testContext.connection
+        );
+        testContext.item = new Label(testContext.connection, {
+          id: '123',
+          displayName: 'foo'
+        });
+        testContext.collection.delete(testContext.item, {'display_name': 'required'});
+        expect(testContext.connection.request).toHaveBeenCalledWith({
+          method: 'DELETE',
+          qs: {},
+          body: {'display_name': 'required'},
+          path: '/labels/123',
+        });
+        done();
+      });
+
+      test('label should populate delete request body if not provided', done => {
```

Comment:
if delete is called with an object that has a displayName, I wanted to populate the request_body with display_name without requiring them to pass it into the method, since it is accessible to the method through the object. Let me try to rename this test to make it more clear what's happening.

Reply:
ahh okay, I thought our API would error out if the displayName wasn't provided but I just checked and it doesn't. 
The test for draft deletion should error out though b/c our docs say:
> Labels can be deleted by issuing a DELETE request to the label’s URI. A label can be deleted even if it still has associated messages.

And I just tried deleting a draft with no request body just to make sure and our API returns the below error:
```
{
    "message": "400 Bad Request: The browser (or proxy) sent a request that this server could not understand.",
    "type": "api_error"
}
```